test-data/unit/check-partially-defined.test

```diff
@@ -137,3 +144,158 @@ else:
     z = 2
 
 a = z + y  # E: Name "y" may be undefined
+[case testReturn]
```

Comment:
Question: will it work well together with rechability check? For example:

```python
if int():
   raise ValueError
   x = 1
else:
   x = 2

a = x  # ?
```

?

Reply:
The example you provided will work. But here's an example where it doesn't work very well yet:
```python
if False:
    y = 1  # E: statement is unreachable
else:
    x = 2

a = x  # E: variable may be undefined
```

The plan is to check if a block is reachable, which I think should make the sample I posted return have only one error (the unreachable one).